src/main/rx/jetiexbus.c

```diff
@@ -0,0 +1,595 @@
+    exMessage[EXTEL_HEADER_DATA] = (labelLength << 3) + unitLength;
+
+    memcpy(&exMessage[EXTEL_HEADER_DATA + 1], Sensor->name, labelLength);
+    memcpy(&exMessage[EXTEL_HEADER_DATA + 1 + labelLength], Sensor->unit, unitLength);
+
+    exMessage[exMessage[EXTEL_HEADER_TYPE_LEN] + 1] = calcCRC8(&exMessage[EXTEL_HEADER_TYPE_LEN], exMessage[EXTEL_HEADER_TYPE_LEN]);
+}
+
+
+uint8_t createExTelemetrieValueMessage(uint8_t *exMessage, uint8_t itemStart)
+{
+    uint8_t lastItem;
+    uint8_t byteCount = 0;
+    uint8_t valueByteCount = 0;
+    uint8_t iCount = 0;
+    uint32_t sensorValue;
+
+
+    if ((itemStart % 16) == 0)
+        itemStart++;
+
+    lastItem = itemStart + 5;
+
+    if (lastItem > JETI_EX_SENSOR_COUNT)
+        lastItem = JETI_EX_SENSOR_COUNT;
+
+    jetiExTelemetry[EXTEL_HEADER_LSN_LB] = itemStart & 0xF0;
+
+    if (itemStart>JETI_EX_SENSOR_COUNT)
+        return 0;
+
+    for (uint8_t item=itemStart; item < lastItem; item++){
+
+        switch(jetiExSensors[item].exDataType) {
+
+        case EX_TYPE_6b:
+            byteCount = 1;
+            break;
+        case EX_TYPE_14b:
+            byteCount = 2;
+            break;
+        case EX_TYPE_22b:
+        case EX_TYPE_DT:
+            byteCount = 3;
+            break;
+        case EX_TYPE_30b:
+        case EX_TYPE_GPS:
+            byteCount = 4;
+            break;
+        }
+
+        exMessage[EXTEL_HEADER_ID + valueByteCount] = ((item & 0xF) << 4) + jetiExSensors[item].exDataType;
+
+
+        sensorValue = jetiExSensors[item].value;
+        if (jetiExSensors[item].value < 0)
+            sensorValue &= resetMask[byteCount - 1];
+
+        sensorValue |= jetiExSensors[item].decimals;
+
+        iCount = 0;
+        do {
+            exMessage[EXTEL_HEADER_DATA + valueByteCount + iCount] = sensorValue;
+            sensorValue = sensorValue>>8;
+            iCount++;
+        } while(iCount < byteCount);
+
+        valueByteCount += 1 + byteCount;
+    }
+
+    exMessage[EXTEL_HEADER_TYPE_LEN] = 0x46+valueByteCount;
+    exMessage[(exMessage[EXTEL_HEADER_TYPE_LEN] & 0x3F) + 1] = calcCRC8(&exMessage[EXTEL_HEADER_TYPE_LEN], (exMessage[EXTEL_HEADER_TYPE_LEN]&0x3F));
+
+    return lastItem;
+}
+
+
+void createExBusMessage(uint8_t *exBusMessage, uint8_t *exMessage, uint8_t exBusID)
+{
+    uint16_t crc16;
+
+    exBusMessage[EXBUS_HEADER_PACKET_ID] = exBusID;
+    exBusMessage[EXBUS_HEADER_SUBLEN] = (exMessage[EXTEL_HEADER_TYPE_LEN] & 0x3F) + 2;    // +2: startbyte & CRC8
+    exBusMessage[EXBUS_HEADER_MSG_LEN] = 8 + exBusMessage[EXBUS_HEADER_SUBLEN];
+
+    crc16 = calcCRC16(exBusMessage,exBusMessage[EXBUS_HEADER_MSG_LEN] - JETIEXBUS_CRC_LEN);
+    exBusMessage[exBusMessage[EXBUS_HEADER_MSG_LEN] - 2] = crc16;
+    exBusMessage[exBusMessage[EXBUS_HEADER_MSG_LEN] - 1] = crc16>>8;
+}
+
+
+void initJetiExBusTelemetry(telemetryConfig_t *initialTelemetryConfig)
+{
+    UNUSED(initialTelemetryConfig);
+
+    // Init Ex Bus Frame header
```

Comment:
It is probably worth the performance penalty to fill these parts when packet is generated. This approach is quite error-prone (and packet data are separated, its hard to read)